buildSrc/src/main/java/org/elasticsearch/gradle/DependenciesGraphTask.java

```diff
@@ -0,0 +1,182 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.gradle.api.GradleException;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.artifacts.DependencySet;
+import org.gradle.api.artifacts.ModuleVersionIdentifier;
+import org.gradle.api.artifacts.ProjectDependency;
+import org.gradle.api.tasks.options.Option;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.TaskAction;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * A task to generate a dependency graph of our runtime dependencies and push that via
+ * an API call to a given endpoint of a SCA tool/service.
+ * The graph is built according to the specification in https://github.com/snyk/dep-graph#depgraphdata
+ *
+ * Due to the nature of our dependency resolution in gradle, we are abusing the aforementioned graph definition as
+ * the graph we construct has a single root ( the subproject ) and all dependencies are children of that root,
+ * irrespective of if they are direct dependencies or transitive ones ( that should be children of other children ).
+ * Although we end up lacking some contextual information, this allows us to scan and monitor only the dependencies
+ * that are bundled and used in runtime.
+ */
+public class DependenciesGraphTask extends DefaultTask {
+
+    private Configuration runtimeConfiguration;
+    private Configuration compileOnlyConfiguration;
+    private String url;
+    private String token;
+
+    @InputFiles
+    public Configuration getRuntimeConfiguration() {
+        return runtimeConfiguration;
+    }
+
+    @InputFiles
+    public Configuration getCompileOnlyConfiguration() {
+        return compileOnlyConfiguration;
+    }
+
+    public void setRuntimeConfiguration(Configuration runtimeConfiguration) {
+        this.runtimeConfiguration = runtimeConfiguration;
+    }
+
+    public void setCompileOnlyConfiguration(Configuration compileOnlyConfiguration) {
+        this.compileOnlyConfiguration = compileOnlyConfiguration;
+    }
+
+    @Option(option = "url", description = "The API endpoint to call with the dependency graph")
+    public void setUrl(String url) {
+        this.url = url;
+    }
+
+    @Input
+    public String getUrl() {
+        return this.url;
+    }
+
+    @Option(option = "token", description = "The API KEY used to authenticate to the API")
```

Comment:
Depending on how we can pass secrets from Jenkins to a gradle task, this can be changed/adapted. I'm up for suggestions on how to deal with this

Reply:
Yeah, we probably dont' want to use a CLI option, as this would leak the secret into the build logs. Given that we can probably ditch the `@Option` bit here and simply read it from something like an environment variable which we can inject into the build.